### ISSUE TITLE
Use author date instead of committer date for lastUpdated

### ISSUE THREAD

**Issue by foxB612** [vuejs/vitepress#2617]
### Is your feature request related to a problem? Please describe.

Currently `lastUpdated` is set to the file's committer date. However, using author date may be more convenient sometimes, for example in these cases:

1. Add a post which was written long time ago but not committed into the repo. In this case the user may want the `lastUpdated` date to be a previous date rather than the current date. This will be possible by `git commit --date= ` if author date is used for `lastUpdated`.
2. Sometimes users may use `git rebase` in their repo, which will refresh committer dates of all involved commits while not actually changing the contents of the files. In this case `lastUpdated` shouldn't be changed IMO.

In summary, if author date is used instead of committer date, users can keep the `lastUpdated` value to be the actual editing date (of the file contents) when managing commits in their repo.

### Describe the solution you'd like

As the title says.

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Read the [Contributing Guidelines](https://github.com/vuejs/vitepress/blob/main/.github/contributing.md).
- [X] Check that there isn't already an issue that asks for the same feature to avoid creating a duplicate.

**Comment by foxB612** [comment]
https://github.com/vuejs/vitepress/blob/789dc16bb134bd06e0ca12b850d57471a3d86ea6/src/node/utils/getGitTimestamp.ts#L8

Currently it's `%ci` which is committer date.

According to documentation of git log pretty formats:

> %ai
>     author date, ISO 8601-like format
> %ci
>     committer date, ISO 8601-like format
